### PR TITLE
x-pack/filebeat/input/httpjson/internal/v2: allow split chains to continue past empty targets

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds a configuration option "ignore_error" that allows a split processor chain to continue if a target field is present but empty.

## Why is it important?

This is required to address #26008 by allowing multiple splits on an input (see https://github.com/elastic/beats/issues/26008#issuecomment-915875217).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~ Defaults to existing behaviour.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Confirm that the new expected split processor results are correct in `split_test.go`

## How to test this PR locally

Standard testing.

## Related issues

- Relates #26008

## Use cases

See related issue.

## Screenshots

N/A

## Logs

N/A

### CODE REVIEW

File: x-pack/filebeat/input/httpjson/internal/v2/split.go [hunk 1]
```diff
@@ -21,18 +21,24 @@ var (
...
+	transforms            []basicTransform
+	child                 *split
+	keepParent            bool
+	ignoreEmptyFieldError bool
+	keyField              string
+	isRoot                bool
+	delimiter             string
 }
 
+// newSplitSplit returns a new split based on the provided config and
```
Comment (marc-gr) [inline x-pack/filebeat/input/httpjson/internal/v2/split.go:40]:
```suggestion
// newSplitResponse returns a new split based on the provided config and
```

File: x-pack/filebeat/docs/inputs/input-httpjson.asciidoc [hunk 2]
```diff
@@ -566,6 +566,11 @@ Required if using split type of `string`.  This is the sub string used to split
 
 Valid when used with `type: map`. When not empty, defines a new field where the original key value will be stored.
 
+[float]
+==== `response.split[].ignore_error`
```
Comment (marc-gr) [inline x-pack/filebeat/docs/inputs/input-httpjson.asciidoc:570]: As it behaves right now, I would expect it to ignore all kind of errors, such as https://github.com/elastic/beats/pull/27880/files#diff-5d902f14f54910145f27c89c42ed581f8c81ffea88cd3cb03cf1f2b86f4ff6a8R127. I am fine with the current behavior of just ignoring empty values, but would suggest then renaming it to something on the lines of `ignore_empty`, similar to other processors. WDYT?

**Review by marc-gr — COMMENTED** [review]
Looks good! Just dropped a thought

File: x-pack/filebeat/input/httpjson/internal/v2/split.go (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/input/httpjson/internal/v2/split.go:40]: Thanks. Fixed.

File: x-pack/filebeat/docs/inputs/input-httpjson.asciidoc (hunk 2 above)
Comment (efd6) [reply x-pack/filebeat/docs/inputs/input-httpjson.asciidoc:570]: I was thinking pretty hard about that, which is why I changed the internal field's name to `ignoreEmptyFieldError`, but you're right. There are errors that it won't ignore as it stands; anything that is non-nil and not `common.ErrKeyNotFound` (which is arguably empty) causes an error return at L106.

There is an `ignore_empty_value` flag used in the httpjson cursor processor and elsewhere, so this is probably a good name to use.

**Review by marc-gr — APPROVED** [review]

**Pull request merged**